cpp/lib/common/TypeDBDriverException.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "typedb/common/TypeDBDriverException.hpp"
+
+namespace TypeDB {
+
+TypeDBDriverException::TypeDBDriverException(const char* code, const char* message)
+    : std::runtime_error(""),
+      errCode(code),
+      errMsg(message) {}
```

Comment:
Copy over the native exception's char* so ownership is simplified.

Reply:
This class should be the one responsible for retrieving the exception contents from over FFI.